server/src/main/java/org/elasticsearch/repositories/blobstore/FileRestoreContext.java

```diff
@@ -190,6 +191,10 @@ private void afterRestore(SnapshotFiles snapshotFiles, Store store, StoreFileMet
                 try {
                     store.deleteQuiet("restore", storeFile);
                     store.directory().deleteFile(storeFile);
+                } catch (ImmutableDirectoryException e) {
```

Comment:
Could we instead up front check that the directory is "immutable" and skip the restore completely? Would feel slightly less hackish to me... Might have to look into a chain of FilterDirectories to see if any are immutable.

Reply:
It's possible but uglier IMO. We can't add a method to `Directory` itself to check that since Lucene owns it, so we'd have to add a marker interface and use `instanceof`.